### DISCUSSION TITLE
v4.0.0

### DISCUSSION THREAD

**Discussion by frederikprijck** [auth0/jwt-decode#233] [Announcements]
We are excited to announce that [a new major version of this library is available in GA](https://github.com/auth0/jwt-decode/releases/tag/v4.0.0), including a couple of improvements:

- No longer include a polyfill for [atob](https://developer.mozilla.org/en-US/docs/Web/API/atob), as this is [supported in all major browsers](https://caniuse.com/?search=atob) (and [node environments > 14](https://developer.mozilla.org/en-US/docs/Web/API/atob#browser_compatibility)).
- Compile to ES2017, dropping support for anything that does not support ES2017 (which should be very limited [according to caniuse](https://caniuse.com/?search=es2017))
- Use Node's atob when running on node.
- Drop support for Node 14, add support for Node 20.
- Add support for package.json's `exports` field, for better CJS/ESM support
- Reorganize build artifacts for better CJS/ESM support (cjs and esm needs to be their own directory with a cjs specific package.json file)
- Drop manual UMD bundle creation in `index.standalone.ts`, but rely on rollup instead.
- Infer JwtPayload and JwtHeader default types from the `header` argument by using overloads.
- Decreased bundle size

Given the amount of downloads we have weekly, as well as the amount of people eventually using this in the apps they use, we are saving gigabytes of traffic per week on the internet with this :shamrock:.

Even though some users might experience breaking changes, mostly because of the `exports` field, the majority should be able to update without making any changes, assuming the SDK is used in environments with support for `atob`.

If you have any questions, need assistance or any feedback, feel free to reach out here on GitHub in the Discussions thread.

Special thanks to @jonkoops for being heavily involved in the release 🚀 🔥 .

**Comment by MrDant** [comment]
Hello is not available for React Native

**Reply by frederikprijck** [reply]
See https://github.com/auth0/jwt-decode/issues/241, see also the [polyfilling section](https://github.com/auth0/jwt-decode#polyfilling-atob) of the README.
